%% bulk_genome_properties.tex
%% Author: Leighton Pritchard
%% Copyright: James Hutton Institute
%% A brief description of useful bulk genome properties

% SUBSECTION: Nucleotide frequency and genome size
\subsection{Nucleotide Frequency/Genome Size}

% Calculating GC content/skew
\begin{frame}
  \frametitle{Nucleotide frequency/genome size}
  \begin{itemize}
    \item Very easy to calculate from complete/draft genome
    \item Can calculate for individual contigs/scaffolds/regions
    \item Usually reported in GUI genome browsers
  \end{itemize}    
  Trivial to determine using, e.g. Python
  \begin{center}
    \includegraphics[width=1\textwidth]{images/python_gc}
  \end{center}  
\end{frame}

% GC content and genome size
\begin{frame}
  \frametitle{Nucleotide frequency/genome size}
  GC content and chromosome size can be characteristic\\
  See \texttt{data/bacteria\_size} for example iPython notebook exercise\\[0.5cm]
  \begin{center}
    \includegraphics[width=1\textwidth]{images/gc_vs_size}
  \end{center}  
\end{frame}

% Blobology
\begin{frame}
  \frametitle{Blobology\footnote{\tiny{\href{http://dx.doi.org/10.1007/s13199-012-0154-6}{Kumar and Blaxter \textit{et al}. (2011) \textit{Symbiosis} \textbf{3}:119-126 doi:10.1007/s13199-012-0154-6}}}}
  \begin{columns}[T]
    \begin{column}{5cm}
      Sequencing samples may be contaminated or contain microbial symbionts.\\
      Expect more host than symbiont/contaminant DNA\\
      GC content and read coverage can be used to separate contigs, following assembly and mapping\\
      \href{http://nematodes.org/bioinformatics/blobology/}{http://nematodes.org/bioinformatics/blobology/}
    \end{column}
    \begin{column}{5cm}
      \includegraphics[width=1\textwidth]{images/blobology}
    \end{column}
  \end{columns}
\end{frame}

% k-mers
\begin{frame}
  \frametitle{$k$-mers}
     \begin{itemize}
       \item Nucleotides: \texttt{[ACGT]}
       \item Dinucleotides: \texttt{[AA|AC|AG|AT|CA|CC|$\ldots$]} (16 dimers)
       \item Trinucleotides: 32 trimers
       \item $k$-mers: $4^k$ $k$-mers
     \end{itemize}
     (see example in \texttt{data/shiny})
\end{frame}

% k-mer figures
\begin{frame}
  \frametitle{$k$-mers}
  Diagnostic differences in $k$-mer frequency, and variability.
  \begin{columns}[T]
    \begin{column}{5cm}
    \textit{E.coli}\\
     \includegraphics[width=1\textwidth]{images/kmer_ecoli}\\
    \end{column}
    \begin{column}{5cm}     
     \textit{Mycoplasma} spp.
     \includegraphics[width=1\textwidth]{images/kmer_mycoplasma}\\
   \end{column}
  \end{columns}
\end{frame}


% SUBSECTION: Nucleotide frequency and genome size
\subsection{Average Nucleotide Identity}

% DNA-DNA hybridisation
\begin{frame}
  \frametitle{DNA-DNA hybridisation\footnote{\tiny{\href{http://dx.doi.org/10.1016/S0168-6445(00)00040-1}{Morello-Mora and Amann (2001) \textit{FEMS Micro. Rev.} \textbf{25}:39-67 doi:10.1016/S0168-6445(00)00040-1}}}}
  \begin{columns}[T]
    \begin{column}{5cm}
      \begin{itemize}
        \item Denature DNA from two organisms.
        \item Allow to anneal. Reassociation $\approx$ similarity, measured as $\Delta T$  of denaturation curves.
        \item Proxy for sequence similarity.
        \item ``Gold Standard'' for prokaryotic taxonomy, since 1960s. ``70\% identity $\approx$ same species.''
      \end{itemize}
    \end{column}
    \begin{column}{5cm}
      \includegraphics[width=1\textwidth]{images/ddh}
    \end{column}
  \end{columns}
\end{frame}

% ANIb
\begin{frame}
  \frametitle{Average Nucleotide Identity (ANIb)\footnote{\tiny{\href{http://dx.doi.org/10.1099/ijs.0.64483-0}{Goris \textit{et al}. (2007) \textit{Int. J. Syst. Biol.} \textbf{57}:81-91 doi:10.1099/ijs.0.64483-0}}}}
  \begin{columns}[T]
    \begin{column}{5cm}
      1. Break genomes into 1020t fragments\\
      2. \textbf{ANIb}: Mean \% identity of all BLASTN matches with $>30\%$ identity and $>70\%$ fragment coverage.\\[0.5cm]
      \begin{itemize}
        \item DDH:ANIb linear
        \item DDH:\%ID linear
        \item 70\%ID $\approx$ 95\%ANIb
      \end{itemize}
    \end{column}
    \begin{column}{5cm}
      \includegraphics[width=1\textwidth]{images/ddh_ani_pid}
    \end{column}
  \end{columns}
\end{frame}

% ANIm
\begin{frame}
  \frametitle{Average Nucleotide Identity (ANIm)\footnote{\tiny{\href{http://dx.doi.org/10.1073/pnas.0906412106}{Richter and Rossello-Mora (2009) \textit{Proc. Natl. Acad. Sci. USA} \textbf{106}:19126-19131 doi:10.1073/pnas.0906412106}}}}
  \begin{columns}[T]
    \begin{column}{3cm}
      1. Align genomes with MUMmer\\
      2. \textbf{ANIm}: Mean \% identity of all matches\\[0.25cm]
      \begin{itemize}
        \item DDH:ANIm linear
        \item 70\%ID $\approx$ 95\%ANIb
      \end{itemize}
    \end{column}
    \begin{column}{7cm}
      \includegraphics[width=1\textwidth]{images/ddh_anim}
    \end{column}
  \end{columns}
  \textbf{TETRA}: tetranucleotide frequency-based classifier introduced in same paper.
\end{frame}

% ANIb/ANIm/TETRA comparison
\begin{frame}
  \frametitle{ANI/TETRA comparison}
  All three methods applied to \textit{Anaplasma} spp.\\[0.25cm]
  \begin{columns}[T]
    \begin{column}{4cm}
    ANIb:\\
      \includegraphics[width=1\textwidth]{images/ANIb}
    \end{column}
    \begin{column}{4cm}
    ANIm:\\
      \includegraphics[width=1\textwidth]{images/ANIm}
    \end{column}
    \begin{column}{4cm}   
    TETRA:\\
      \includegraphics[width=1\textwidth]{images/TETRA}
    \end{column}
  \end{columns}       
  ANIb discards information, relative to ANIm\\
  ANIb/ANIm reflect evolutionary history; TETRA reflects bulk composition
\end{frame}

% ANIm in practice
\begin{frame}
  \frametitle{ANI/TETRA comparison}
  Two practical applications\\[0.25cm]
  \begin{columns}[T]
    \begin{column}{5cm}
    29 \textit{Dickeya} isolates:\\
      \includegraphics[width=1\textwidth]{images/ANIm_Ecoli}
    \end{column}
    \begin{column}{5cm}
    180 \textit{E.coli} isolates:\\
      \includegraphics[width=1\textwidth]{images/ANIm}
    \end{column}
  \end{columns}       
  ANIb discards information, relative to ANIm\\
  ANIb/ANIm reflect evolutionary history; TETRA reflects bulk composition
\end{frame}