%% high_throughput_sequencing.tex
%% Author: Leighton Pritchard
%% Copyright: James Hutton Institute
%% These slides briefly introduce the four main sequencing technologies
%% and comment on their relative performance and price.
%% Next they mention Oxford Nanopore as the next promising technology,
%% and the potential impact it could have on sequencing.
%% Finally, the impact that this technology has had on databases and 
%% analyses etc. are described.

% SUBSECTION: Four dominant technologies
% A summary of the characteristics of the four dominant sequencing 
% technologies, and how they can be benchmarked
\subsection{Three revolutions, four dominant technologies}

% The first two of three revolutions
\begin{frame}
  \frametitle{Revolutions One and Two\footnote{\tiny{Loman and Pallen (2015) \textit{Nat. Rev. Micro.} \href{http://dx.doi.org/10.1038/nrmicro3565}{doi:10.1038/nrmicro3565}}}}
  \textbf{Revolution One:} whole-genome shotgun
  \begin{itemize}
    \item First bacterial genomes: \textit{Haemophilus influenzae} (1995); \textit{E. coli}, \textit{Bacillus subtilis} (1997)
    \item (Oh, and the human genome)
  \end{itemize}
  \textbf{Revolution Two:} high-throughput sequencing
  \begin{itemize}
    \item "Next-generation" sequencing (now "last-generation").
    \item 454 GS20 (2005), Illumina GAII (2007).
    \item metagenomics; surveillance sequencing; SNP-based comparisons; transposon-sequencing for functional genomics; ChIP-seq; $\ldots$
  \end{itemize}
\end{frame}

% The four main current sequencing technologies
\begin{frame}
  \frametitle{Not all HT sequencing is the same}
  It's all about the biology, but it all starts with the data.\\
  Sequencing technology (including library prep.) affects your sequence data.
  \begin{itemize}
    \item Roche/454 \includegraphics[height=0.15\textheight]{images/454_sequencer}
    \item Illumina \includegraphics[height=0.15\textheight]{images/miseq_nextseq_hiseq}
    \item Ion Torrent \includegraphics[height=0.15\textheight]{images/ion_torrent_personal_genome_maker}
    \item Pacific Bioscience (PacBio) \includegraphics[height=0.15\textheight]{images/PacBio_RSII}
  \end{itemize}
\end{frame}

% Basic principle
\begin{frame}
  \frametitle{The basic principle}
  DNA source is fragmented, and the fragments are sequenced. \\
  \begin{center}
    \includegraphics[height=0.7\textheight]{images/library_preparation}
  \end{center}      
\end{frame}

% Paired- and single-end reads
\begin{frame}
  \frametitle{HTS: PE vs SE}
  High-throughput sequencing (e.g. Illumina), reads may be single-end, or paired-end.
  \begin{center}
    \includegraphics[height=0.55\textheight]{images/pe_vs_se}
  \end{center}      
  Putting the jigsaw back together is sequence assembly.
\end{frame}

% Broad differences in chemistry and output
\begin{frame}
  \frametitle{Four different chemistries\footnote{\tiny{Loman \textit{et al}. (2012) \textit{Nat. Rev. Micro.} \textbf{31}:294-296 \href{http://dx.doi.org/10.1038/nbt.2522}{doi:10.1038/nbt.2522}}}}
  Reads differ by technology, and may require different bioinformatic treatment$\ldots$
  \begin{itemize}
    \item \textbf{Roche/454}: Pyrosequencing (long reads, but expensive, and high homopolymer errors) (700-800bp, 0.7Gbp, 23h)
    \item \textbf{Illumina}: Reversible terminator (cost-effective, massive throughput, but short read lengths) (2x150bp, 1.5Gbp, 27h)
    \item \textbf{Ion Torrent}: Proton detection (short run times, good throughput, high homopolymers errors) (200bp, 1Gbp, 3h)
    \item \textbf{PacBio}: Real-time sequencing (very long reads, high error rate, expensive) (3-15kbp, 3Gbp/day, 20min)
  \end{itemize}
  $\ldots$ different error profiles, varying capability to assemble/determine variation
\end{frame}

% Sequencer relative costs
\begin{frame}
  \frametitle{Costs of sequencing\footnote{\tiny{Miyamoto \textit{et al}. (2014) \textit{BMC Genomics} \textbf{15}:699 \href{http://dx.doi.org/10.1186/1471-2164-15-699}{doi:10.1186/1471-2164-15-699}}}}
    \begin{center}
      \includegraphics[width=1\textwidth]{images/miyamoto_costs}
    \end{center}      
\end{frame}

% SUBSECTION: Benchmarking
% Summary of benchmarking for sequencing methods: which is best?
\subsection{Benchmarking}

% Benchmarking is very useful
\begin{frame}
  \frametitle{Benchmarked performance}
  Apply several sequencing technologies to the same sample(s). \\  
  Benchmark comparisons inform appropriate choice of sequencing technology%
\footnote{\tiny{Miyamoto \textit{et al}. (2014) \textit{BMC Genomics} \textbf{15}:699 \href{http://dx.doi.org/10.1186/1471-2164-15-699}{doi:10.1186/1471-2164-15-699}}}$^{,}$%
\footnote{\tiny{Salipante \textit{et al}. (2014) \textit{Appl. Environ. Micro.} \textbf{80}:7583-7591 \href{http://dx.doi.org/10.1128/AEM.02206-14}{doi:10.1128/AEM.02206-14}}}$^{,}$%
\footnote{\tiny{Frey \textit{et al}. (2014) \textit{BMC Genomics} \textbf{15}:96 \href{http://dx.doi.org/10.1186/1471-2164-15-96}{doi:10.1186/1471-2164-15-96}}}$^{,}$%
\footnote{\tiny{Koshimizu \textit{et al}. (2013) \textit{PLoS One} \textbf{8}:e74167 \href{http://dx.doi.org/10.1371/journal.pone.0074167}{doi:10.1371/journal.pone.0074167}}}$^{,}$%
\footnote{\tiny{Quail \textit{et al}. (2012) \textit{BMC Genomics} \textbf{13}:341 \href{http://dx.doi.org/10.1186/1471-2164-13-341}{doi:10.1186/1471-2164-13-341}}}$^{,}$%
\footnote{\tiny{Loman \textit{et al}. (2012) \textit{Nat. Biotech.} \textbf{30}:434-439 \href{http://dx.doi.org/10.1038/nbt.2198}{doi:10.1038/nbt.2198}}}$^{,}$%
\footnote{\tiny{Lam \textit{et al}. (2011) \textit{Nat. Biotech.} \textbf{1} (6) \href{http://dx.doi.org/10.1038/nbt.2065}{doi:10.1038/nbt.2065}}} \\[0.5cm]
  Progress in technologies is driving research very rapidly. \\
  Always look for most recent/relevant benchmarks. \\[0.5cm]
  \textbf{Bioinformatic methods also need to be benchmarked.}
\end{frame}

% Benchmarking methods
\begin{frame}
  \frametitle{Benchmarking on \textit{Vibrio}\footnote{\tiny{Miyamoto \textit{et al}. (2014) \textit{BMC Genomics} \textbf{15}:699 \href{http://dx.doi.org/10.1186/1471-2164-15-699}{doi:10.1186/1471-2164-15-699}}}}
    \begin{itemize}
      \item Sequenced \textit{Vibrio parahaemolyticus} (2x chromosomes, closed reference genome) with four technologies
      \item Chose an assembler for each tech, and assembled reads
      \item Excess reads with Ion/MiSeq: used random subsets of reads to determine required coverage 
      \item Aligned assemblies (MUMmer) to known high-quality chromosome sequence, to measure error
    \end{itemize}      
    \begin{center}
      \includegraphics[width=0.8\textwidth]{images/miyamoto_errors}
    \end{center}      
\end{frame}

% Benchmarking results: raw data
\begin{frame}
  \frametitle{Benchmarking on \textit{Vibrio}\footnote{\tiny{Miyamoto \textit{et al}. (2014) \textit{BMC Genomics} \textbf{15}:699 \href{http://dx.doi.org/10.1186/1471-2164-15-699}{doi:10.1186/1471-2164-15-699}}}}
    \begin{center}
      \includegraphics[height=0.8\textheight]{images/miyamoto_table}
    \end{center}      
\end{frame}

% Benchmarking results: alignments
\begin{frame}
  \frametitle{Benchmarking on \textit{Vibrio}\footnote{\tiny{Miyamoto \textit{et al}. (2014) \textit{BMC Genomics} \textbf{15}:699 \href{http://dx.doi.org/10.1186/1471-2164-15-699}{doi:10.1186/1471-2164-15-699}}}}
    \textit{De novo} assembly and alignment against \textit{Vibrio parahaemolyticus} (2x chromosomes)
    \begin{center}
      \includegraphics[width=1\textwidth]{images/miyamoto_alignments}
    \end{center}      
\end{frame}

% Benchmarking results: conclusions
\begin{frame}
  \frametitle{Benchmarking on \textit{Vibrio}\footnote{\tiny{Miyamoto \textit{et al}. (2014) \textit{BMC Genomics} \textbf{15}:699 \href{http://dx.doi.org/10.1186/1471-2164-15-699}{doi:10.1186/1471-2164-15-699}}}}
  \begin{itemize}
    \item More and longer reads do not always give the best assemblies: read depth, read distribution, error rate also matters
    \item Optimal assemblies were obtained at around 60x-80x coverage, for Illumina and Ion.
    \item Multiple rRNA regions are fragmented in short-read assemblies
    \item PacBio generated single chromosome contigs
    \item Assembly of multiple-chromosome bacteria is currently feasible
  \end{itemize}  
  Variability in published genomes as methods are not standard (e.g. sequencing technology, assembler, parameter settings and pre-processing)$\ldots$
\end{frame}

% SUBSECTION: Nanopore
% This is likely to become the dominant sequencing technology
\subsection{Nanopore}

% The third revolution
\begin{frame}
  \frametitle{Revolution Three\footnote{\tiny{Loman and Pallen (2015) \textit{Nat. Rev. Micro.} \href{http://dx.doi.org/10.1038/nrmicro3565}{doi:10.1038/nrmicro3565}}}}
  \textbf{Revolution Three:} single-molecule long-read sequencing
  \begin{itemize}
    \item Living through the revolution, now
    \item PacBio (SMRT): large machine, expensive
    \item Nanopore: portable device, inexpensive
    \item Less mature, less accurate, improving rapidly
  \end{itemize}
\end{frame}

% Nanopore
\begin{frame}
  \frametitle{The future dominant sequencer?}
  Oxford Nanopore. A sequencer the size of your hand.
    \begin{center}
      \includegraphics[width=0.48\textwidth]{images/minion_run}\thinspace
      \includegraphics[width=0.48\textwidth]{images/nanopore_schematic}
    \end{center} 
    \begin{itemize}
      \item Microfluidics, single-molecule sequencing; 11-70kbp reads
      \item Reports current across pore (tiny electron microscope) as molecule moves through
      \item \$10/Mbp, 110Mbp per flowcell\footnote{\tiny{Yaniv Erlich (2013) \href{http://erlichya.tumblr.com/post/66376172948/hands-on-experience-with-oxford-nanopore-minion}{Future Continuous blog}}}
    \end{itemize}          
\end{frame}

% Some better results
\begin{frame}
  \frametitle{Early data\footnote{\tiny{Quick \textit{et al}. (2014) \textit{GigaScience} \textbf{3}:22 \href{http://dx.doi.org/10.1111/1755-0998.12324}{doi:10.1111/1755-0998.12324}}}}
  It's a fast-moving area, and results are improving.
    \begin{center}
      \includegraphics[height=0.65\textheight]{images/nanopore_quick}
    \end{center}           
\end{frame}

% Some better results
\begin{frame}
  \frametitle{Developing tools}
  Oxford Nanopore's open beta went out without analysis tools.\\
  Tools (Poretools, poRe, etc.) were written/tested/validated by the user community%
  \footnote{\tiny{Loman and Quinlan (2014) \textit{Bioinformatics} \href{http://dx.doi.org/10.1093/bioinformatics/btu555}{doi:10.1093/bioinformatics/btu555}}}$,$%
  \footnote{\tiny{Watson \textit{et al}. (2014) \textit{Bioinformatics} \href{http://dx.doi.org/10.1093/bioinformatics/btu590}{doi:10.1093/bioinformatics/btu590}}}
    \begin{center}
      \includegraphics[height=0.55\textheight]{images/poretools}
    \end{center}           
\end{frame}

% Some recent applications
\begin{frame}
  \frametitle{Recent applications}
  \begin{itemize}
    \item Amplicon sequencing (16S metagenomics) of bacteria and viruses \footnote{\tiny{Kilianski \textit{et al}. (2015) \textit{GigaScience} \href{http://dx.doi.org/10.1186/s13742-015-0051-z}{doi:10.1186/s13742-015-0051-z}}}
    \item Real-time viral diagnostics \footnote{\tiny{Greninger \textit{et al}. (2015) \textit{Genome Med.} \href{http://dx.doi.org/10.1186/s13073-015-0220-9}{doi:10.1186/s13073-015-0220-9}}}
    \item Scaffolding of a bacterial genome \footnote{\tiny{Karlsson \textit{et al}. (2015) \textit{Sci. Reports} \href{http://dx.doi.org/10.1038/srep11996}{doi:10.1038/srep11996}}}
    \item Complete \textit{de novo} assembly of a bacterial genome \footnote{\tiny{Loman \textit{et al}. (2015) \textit{Nat. Meth.} \href{http://dx.doi.org/10.1038/nmeth.3444}{doi:10.1038/nmeth.3444}}}
  \end{itemize}
    \begin{center}
      \includegraphics[height=0.3\textheight]{images/minion_asm_dotplot}
      \includegraphics[height=0.3\textheight]{images/minion_asm_tablet}
    \end{center}           
\end{frame}


% Three revolutions summary
\begin{frame}
  \frametitle{The three revolutions\footnote{\tiny{Loman and Pallen (2015) \textit{Nat. Rev. Micro.} \href{http://dx.doi.org/10.1038/nrmicro3565}{doi:10.1038/nrmicro3565}}}}
    \begin{center}
      \includegraphics[width=0.9\textwidth]{images/revolutions}\thinspace
    \end{center} 
\end{frame}

% SUBSECTION: How fast is data increasing?
% Summary of how fast the amount of sequence data is increasing
\subsection{How fast is sequence data increasing?}

% How many are we going to have?
\begin{frame}
  \frametitle{Predicting the future is hard$\ldots$}
    ``How many genomes will we have, and when?'' \\
    Su \textit{et al}. attempted to answer this\footnote{\tiny{\href{http://sulab.org/2013/06/sequenced-genomes-per-year/}{http://sulab.org/2013/06/sequenced-genomes-per-year/}}}:
    \begin{center}
      \includegraphics[width=0.5\textwidth]{images/cumulative_sequenced_genomes1}
      \includegraphics[width=0.5\textwidth]{images/cumulative_sequenced_genomes2}
    \end{center}     
%    How will we keep this much genomic data well-organised?
\end{frame}

% But how many genomes do we have, now?
\begin{frame}
  \frametitle{After that, the flood$\ldots$}
  High-throughput sequencing methods have completely changed the landscape of microbiology \\
  \textbf{(Nearly) complete, (mainly) accurate sequence data is now inexpensive} (and cheaper than analysis)
  \begin{itemize}
    \item \href{http://www.genomesonline.org/cgi-bin/GOLD/index.cgi?page_requested=Complete+Genome+Projects&subset_requested=Complete+And+Published}{GOLD} (19/2/2014): 3,011 ``finished'' ; 9,891 ``permanent draft'' genomes
    \item \href{http://www.genomesonline.org/cgi-bin/GOLD/index.cgi?page_requested=Complete+Genome+Projects&subset_requested=Complete+And+Published}{GOLD} (10/11/2015): 7,657 ``finished'' ; 27,438 ``permanent draft'' genomes; 50,673 prokaryotes
    \item \href{http://www.ncbi.nlm.nih.gov/Traces/wgs/}{NCBI WGS} (19/2/2014): 17,023 microbial genomes
    \item \href{http://www.ncbi.nlm.nih.gov/genome/browse/}{NCBI Genome} (10/11/2015): 55,033 prokaryotic genomes
  \end{itemize}
\end{frame}

% A specific example: Pseudomonas
\begin{frame}
  \frametitle{\textit{Pseudomonas}}
    In 2011, 25 isolate sequences\footnote{\tiny{Studholme (2011) \textit{Mol. Plant Pathol.} \href{http://dx.doi.org/10.1111/j.1364-3703.2011.00713.x}{doi:10.1111/j.1364-3703.2011.00713.x}}}; in 2015, 2098 genomes:
    \begin{center}
      \includegraphics[height=0.5\textheight]{images/pseud2011_abstract}
      \includegraphics[height=0.5\textheight]{images/pseud2015_genomecount}
    \end{center}     
    We're going to need bigger bioinformatics$\ldots$
%    How will we keep this much genomic data well-organised?
\end{frame}

% SUBSECTION: Genome sequence standard
% Brief account of MIGS
%\subsection{Genome Sequence and Data Standards}
%
%% MIGS
%\begin{frame}
%  \frametitle{MIGS\footnote{\tiny{Field \textit{et al}. (2008) \textit{Nat. Biotech.} \textbf{26}:541-547 \href{http://dx.doi.org/10.1038/nbt.1823}{doi:10.1038/nbt.1823}}}}
%  Minimum Information about a (Meta)Genome Sequence (MIGS) \\
%  \begin{itemize}
%    \item Standardisation of the data describing a genome sequence; standards are \textbf{essential} to data integrity and reuse
%    \item Defines core descriptors of the data that cannot be derived from the data (geographical context, experimental methods, etc.)
%    \item Produced by \href{http://en.wikipedia.org/wiki/Genomic_Standards_Consortium}{Genomic Standards Consortium} (\href{http://gensc.org/}{gensc.org})
%    \item Provided as checklist and XML schema
%    \item Now also ``Minimum Information about an ENvironmental Sequence'' (MIENS) and others\footnote{\tiny{Yilmaz \textit{et al}. (2011) \textit{Nat. Biotech.} \textbf{29}:541-547 \href{http://dx.doi.org/10.1038/nbt1360}{doi:10.1038/nbt1360}}}
%    \item Global Microbial Identifier (\href{http://www.globalmicrobialidentifier.org/}{http://www.globalmicrobialidentifier.org/})
%  \end{itemize}        
%\end{frame}
%
%% MIGS table
%\begin{frame}
%  \frametitle{MIGS\footnote{\tiny{Field \textit{et al}. (2008) \textit{Nat. Biotech.} \textbf{26}:541-547 \href{http://dx.doi.org/10.1038/nbt.1823}{doi:10.1038/nbt.1823}}}}
%  Minimum Information about a (Meta)Genome Sequence (MIGS) \\
%  \begin{center}
%    \includegraphics[width=1\textwidth]{images/migs_table}
%  \end{center}           
%\end{frame}